src/errors.rs

```diff
@@ -8,8 +8,31 @@ use crate::protocol::Error;
 pub enum ParseError {
     /// Not enough data was provided.
     InsufficientData,
-    /// Another error while parsing some data.
-    ParseError,
+
+    /// A value did not fit.
+    ///
+    /// This error can e.g. happen when a value that was received from the X11 server does not fit
+    /// into an `usize`.
+    ConversionFailed,
```

Comment:
Can this error ever actually happen? The primary use I see is converting u8s and u32s to usizes to call parse_list and that conversion will always succeed because usize is at least 32 bits.

Reply:
Well... in theory: yes, of course this can happen. ;-)

In practice... Hm...

Why do you think that `usize` is guaranteed to be at least 32 bits?

There are just implementations of `From<u8>` and `From<u16>` for `usize`, so I guess the Rust devs want to keep the possibility of 16 bit targets. Even with that I cannot find some piece of code that actually needs this.

This patch to the code generator:
```diff
diff --git a/generator/src/generator/namespace.rs b/generator/src/generator/namespace.rs
index dca9931..42fcede 100644
--- a/generator/src/generator/namespace.rs
+++ b/generator/src/generator/namespace.rs
@@ -3909,7 +3909,7 @@ impl<'ns, 'c> NamespaceGenerator<'ns, 'c> {
                         outln!(
                             out,
                             "let ({}, remaining) = crate::x11_utils::parse_u8_list({}, \
-                             {}.try_into().or(Err(ParseError::ConversionFailed))?)?;",
+                             {}.into())?;",
                             rust_field_name,
                             from,
                             self.expr_to_str(
@@ -3951,7 +3951,7 @@ impl<'ns, 'c> NamespaceGenerator<'ns, 'c> {
                     outln!(
                         out,
                         "let ({}, remaining) = crate::x11_utils::parse_list::<{}>(remaining, \
-                         {}.try_into().or(Err(ParseError::ConversionFailed))?)?;",
+                         {}.into())?;",
                         rust_field_name,
                         rust_element_type,
                         self.expr_to_str(
@@ -3985,7 +3985,7 @@ impl<'ns, 'c> NamespaceGenerator<'ns, 'c> {
                         outln!(
                             out,
                             "let list_length = \
-                             usize::try_from({}).or(Err(ParseError::ConversionFailed))?;",
+                             usize::from({});",
                             self.expr_to_str(
                                 length_expr,
                                 to_rust_variable_name,
@@ -4051,7 +4051,7 @@ impl<'ns, 'c> NamespaceGenerator<'ns, 'c> {
 
                 outln!(
                     out,
-                    "let fds_len = usize::try_from({}).or(Err(ParseError::ConversionFailed))?;",
+                    "let fds_len = usize::from({});",
                     self.expr_to_str(
                         &fd_list_field.length_expr,
                         to_rust_variable_name,
```
causes 265 compiler errors.

E.g. `xinerama`'s `QueryScreens` request has this:
```xml
            <field type="CARD32" name="number" />
            <pad bytes="20" />
            <list type="ScreenInfo" name="screen_info">
                <fieldref>number</fieldref>
            </list>
```
This would fail on 16 bit targets. In theory. In weird situations.

From a quick look, all the errors seem to be about `u32` conversion (of course, this is expected).